src/main/scala/beam/agentsim/infrastructure/taz/TAZTreeMap.scala

```diff
@@ -125,9 +128,22 @@ class TAZTreeMap(val tazQuadTree: QuadTree[TAZ], val useCache: Boolean = false)
 
   def mapNetworkToTAZs(network: Network): Unit = {
     if (tazListContainsGeoms) {
+      val extent =
+        (tazQuadTree.getMinEasting, tazQuadTree.getMinNorthing, tazQuadTree.getMaxEasting, tazQuadTree.getMaxNorthing)
```

Comment:
The 2e3 buffer should be done here. I see no need to spin things up continuously. Also, the updating of the `tazToLinkIdMapping` can be done in the `match` directly. No need to toss variables around that aren't utilized other than that.

Reply:
I don't quite follow here. I think one reason to loop over all TAZs first is to ensure that `idToTAZMapping` has an entry for each TAZ (even if the network doesn't have any links in that TAZ) so that I can do a `.get(tazId)` later without worrying about a missing key. Any suggestions for simplifying taking that into account?

I added constants distinguishing between the buffers for each TAZ and for the entire map. I think the map buffer should be included in e.g. `tazQuadTree.getMinEasting`, but I want to be sure to also include a buffer in the TAZ-specific treeMap I'm creating to keep track of link locations

Reply:
You actually changed the logic in this recent change. Previously you had subtracted/added 2e3 to the min/max no matter if it was `extent` or from the `env`. In the latest change you only do it if it is from `env`. In the new code my suggestion is something more like this (note that `extent` is reused):

```
val extent =
       new QuadTree[Link](tazQuadTree.getMinEasting, tazQuadTree.getMinNorthing, tazQuadTree.getMaxEasting, tazQuadTree.getMaxNorthing)
    idToTAZMapping.toList.foreach { case (id, taz) =>
        taz.geometry.map(_.getEnvelope.getEnvelopeInternal) match {
          case Some(env) => tazToLinkIdMapping(id) = new QuadTree[Link](
              env.getMinX - tazBoundingBoxBufferMeters,
              env.getMinY - tazBoundingBoxBufferMeters,
              env.getMaxX + tazBoundingBoxBufferMeters,
              env.getMaxY + tazBoundingBoxBufferMeters
            )
          case _ => tazToLinkIdMapping(id) = extent
        }
      }
```

In the old code it would be more like this (same as above, but with the buffer added to the extent):

```
val extent =
       new QuadTree[Link](tazQuadTree.getMinEasting - tazBoundingBoxBufferMeters, tazQuadTree.getMinNorthing - tazBoundingBoxBufferMeters, tazQuadTree.getMaxEasting + tazBoundingBoxBufferMeters, tazQuadTree.getMaxNorthing + tazBoundingBoxBufferMeters)
      idToTAZMapping.toList.foreach { case (id, taz) =>
        taz.geometry.map(_.getEnvelope.getEnvelopeInternal) match {
          case Some(env) =>tazToLinkIdMapping(id) = new QuadTree[Link](
              env.getMinX - tazBoundingBoxBufferMeters,
              env.getMinY - tazBoundingBoxBufferMeters,
              env.getMaxX + tazBoundingBoxBufferMeters,
              env.getMaxY + tazBoundingBoxBufferMeters
            )
          case _         => tazToLinkIdMapping(id) = extent
        }
    }
```

Ultimately my original comment was that extent could be initialized once and then reused over and over, rather than being built, extracted, and rebuilt.